src/tallies/tally.cpp

```diff
@@ -240,37 +240,295 @@ score_str_to_int(std::string score_str)
 // Tally object implementation
 //==============================================================================
 
-Tally::Tally()
+Tally::Tally(int32_t id)
+  : index_{model::tallies.size()}
 {
-  this->set_filters(nullptr, 0);
+  this->set_id(id);
+  this->set_filters({});
 }
 
-void
-Tally::init_from_xml(pugi::xml_node node)
+Tally::Tally(pugi::xml_node node)
+  : index_{model::tallies.size()}
 {
+  // Copy and set tally id
+  if (!check_for_node(node, "id")) {
+    throw std::runtime_error{"Must specify id for tally in tally XML file."};
+  }
+  int32_t id = std::stoi(get_node_value(node, "id"));
+  this->set_id(id);
+
   if (check_for_node(node, "name")) name_ = get_node_value(node, "name");
+
+  // =======================================================================
+  // READ DATA FOR FILTERS
+
+  // Check if user is using old XML format and throw an error if so
+  if (check_for_node(node, "filter")) {
+    throw std::runtime_error{"Tally filters must be specified independently of "
+      "tallies in a <filter> element. The <tally> element itself should "
+      "have a list of filters that apply, e.g., <filters>1 2</filters> "
+      "where 1 and 2 are the IDs of filters specified outside of "
+      "<tally>."};
+  }
+
+  // Determine number of filters
+  std::vector<int> filter_ids;
+  if (check_for_node(node, "filters")) {
+    filter_ids = get_node_array<int>(node, "filters");
+  }
+
+  // Allocate and store filter user ids
+  std::vector<Filter*> filters;
+  if (!filter_ids.empty()) {
```

Comment:
This outer if statement is redundant.

Reply:
Indeed it is. Thanks for pointing that out!